### PR TITLE
Enable eslint:recommended and plugin/node/recommended

### PR DESCRIPTION
**Pull request by sehrope**
Couple of commits to progressively enable tighter eslint config via adding missing strict tag and disabling rule violations locally via comments. Also removes a couple unused eslint plugin dependencies.

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -1,6 +1,19 @@
...
+  "parserOptions": {
+    "ecmaVersion": 2017
+  },
+  "env": {
+    "node": true,
+    "es6": true
+  },
   "rules": {
-    "no-new-func": "off"
+    "no-unused-vars": 0
```
Comment (charmander) [inline .eslintrc]: Are there a lot of these for good reasons?

Comment (sehrope) [reply .eslintrc]: Not really. If you enable it there's a handful of event handlers that don't use the `msg` parameter:

```text
lib/client.js
  283:39  error  'msg' is defined but never used  no-unused-vars
  288:34  error  'msg' is defined but never used  no-unused-vars
  300:37  error  'msg' is defined but never used  no-unused-vars
  306:38  error  'msg' is defined but never used  no-unused-vars

lib/query.js
  220:49  error  'connection' is defined but never used  no-unused-vars
```

I think I held off to keep the incremental diffs minimal and not touch the event handlers. Lemme take a peek as I think it's just a couple ignore tag comments to enable it and would be nice going forward.

**Comment by sehrope** [comment]
@charmander I added the ignore tags. It's only a couple of places and seems better to keep all the event handlers defined with the same signature rather than removing the unused params on just those.

**Comment by sehrope** [comment]
Test failure seems to be something unrelated: https://travis-ci.org/brianc/node-postgres/jobs/517418718#L730-L741

```
type-coercion-tests.js
  test type coercion integer ✔
  test type coercion smallint ✔
  test type coercion bigint ✔
  test type coercion varchar(5) ✔
  test type coercion oid ✔
  test type coercion bool ✔
  test type coercion numeric ✔
  test type coercion real ✔
  test type coercion double precision ✔
  test type coercion timestamptz ✔
  test type coercion timestamp ✔
  test type coercion timetz ✔
  test type coercion time ✔
  timestampz round trip ✔
  selecting nulls ✔
  date range extremes Message: -4712 == -4713
AssertionError: -4712 == -4713
    at /home/travis/build/brianc/node-postgres/test/integration/client/type-coercion-tests.js:182:14
    at Query.<anonymous> (/home/travis/build/brianc/node-postgres/test/test-helper.js:101:7)
    at Query.<anonymous> (/home/travis/build/brianc/node-postgres/test/test-helper.js:148:16)
    at Query.handleReadyForQuery (/home/travis/build/brianc/node-postgres/lib/query.js:125:10)
    at Connection.<anonymous> (/home/travis/build/brianc/node-postgres/lib/client.js:215:19)
    at emitOne (events.js:96:13)
    at Connection.emit (events.js:188:7)
    at Socket.<anonymous> (/home/travis/build/brianc/node-postgres/lib/connection.js:125:12)
    at emitOne (events.js:96:13)
```

I wonder if there's some kind of timezone update / change on newer version of the servers that's breaking that test. Don't recall seeing it before and quick look at the latest release notes only shows the usual "*TZ data updated...*"

**Comment by sehrope** [comment]
Rebased on the latest and passing Travis now that #1867 is merged.

**Comment by brianc** [comment]
sweet! Merging now!

**Pull request merged**